File: 27.remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int count = 0;
        for (int i = 0; i < nums.length-count; i++) {
            if (nums[i] == val) {
                while (nums[nums.length-count-1] == val) {
                    if (nums.length-count-1 <= i) return nums.length-count-1;
                    count++;
                }
                nums[i] = nums[nums.length-count-1];
                count++;
            }
        }
        return nums.length - count;
    }
}

File: Python/27-remove-element.py
from typing import List

class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        val_count = 0
        for index in range(len(nums)):
            if nums[index] != val:
                nums[index - val_count] = nums[index]
            else:
                val_count += 1
        return len(nums) - val_count